### PR TITLE
add flip animation to ZZ logo when switching partner mode

### LINKED ISSUE #1591: Flip animation on ZZ logo when switching to partner
**Is your feature request related to a problem? Please describe.**
It's very nice to feel/see something changes when switching to partner mode.

**Describe the solution you'd like**
A quick flip or turn animation when switching to partner mode. 

**Describe alternatives you've considered**
None.

**Additional context**
Can probably be done with standard css animations.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1591

### CODE REVIEW

**Comment by tijlleenders** [comment]
Looks awesome @vinaybadgujar102 !

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -65,6 +67,11 @@ const Header: React.FC<IHeader> = ({ title, debounceSearch }) => {
     handlePopState();
   }, [location]);
 
+  useEffect(() => {
+    const timer = isFlipping ? setTimeout(() => setIsFlipping(false), 500) : undefined;
+    return () => clearTimeout(timer);
+  }, [isFlipping, setIsFlipping]);
```
Comment (Tushar-4781) [inline src/components/Header/Header.tsx:73]: @vinaybadgujar102 please correct me if i m wrong
Including setIsFlipping in the dependency array might seem redundant in this context. When setIsFlipping is called, it inherently triggers a change in the isFlipping state right ?

File: src/store/index.ts [hunk 2]
```diff
@@ -95,4 +95,9 @@ export const focusTaskTitle = atom({
   default: "",
 });
 
+export const flipAnimationState = atom({
```
Comment (Tushar-4781) [inline src/store/index.ts:98]: why we are making it a global state ? and not use it as local state?

**Review by Tushar-4781 — COMMENTED** [review]
Nicely done @vinaybadgujar102 
Can you please resolve my 2 queries before we merge this ?

File: src/store/index.ts (hunk 2 above)
Comment (vinaybadgujar102) [reply src/store/index.ts:98]: if we use local state, the header component re-renders when handlePartner() is called which resets the flipping state to false

File: src/components/Header/Header.tsx (hunk 1 above)
Comment (vinaybadgujar102) [reply src/components/Header/Header.tsx:73]: agreed, including "setIsFlipping" is redundant

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**